### PR TITLE
Add human readable page title reference

### PR DESCRIPTION
**Pull request by ddouglasz**
Add human readable page title override  reference in the starter kit.

### CODE REVIEW

File: application-templates/starter/src/components/channel-details/channel-details.jsx [hunk 1]
```diff
@@ -84,55 +85,74 @@ const ChannelDetails = (props) => {
...
-            isPrimaryButtonDisabled={
-              formProps.isSubmitting || !formProps.isDirty || !canManage
+    <>
+      <ApplicationPageTitle
+        additionalParts={[
+          `${formatLocalizedString(
+            {
+              name: transformLocalizedFieldToLocalizedString(
+                channel?.nameAllLocales ?? []
+              ),
```
Comment (emmenko) [inline application-templates/starter/src/components/channel-details/channel-details.jsx:95]: This is already done in the form values, so we don't need to do it again.

I would suggest to render the `<ApplicationPageTitle>` within the `<FormModalPage>` children and reuse the same value used for the `title` prop (you want to extract it into a variable).

See below

File: application-templates/starter/src/components/channel-details/channel-details.jsx [hunk 2]
```diff
@@ -84,55 +85,74 @@ const ChannelDetails = (props) => {
...
+                <ContentNotification type="error">
+                  <Text.Body>
+                    {intl.formatMessage(messages.channelDetailsErrorMessage)}
+                  </Text.Body>
+                </ContentNotification>
+              )}
+              {channel && formProps.formElements}
+              {channel === null && <PageNotFound />}
+            </FormModalPage>
+          );
```
Comment (emmenko) [inline application-templates/starter/src/components/channel-details/channel-details.jsx:152]:
```suggestion
          const channelName = formatLocalizedString(
            {
              name: formProps.values?.name,
            },
            {
              key: 'name',
              locale: dataLocale,
              fallbackOrder: projectLanguages,
              fallback: NO_VALUE_FALLBACK,
            }
          );
          return (
            <FormModalPage
              title={channelName}
              isOpen
              onClose={props.onClose}
              isPrimaryButtonDisabled={
                formProps.isSubmitting || !formProps.isDirty || !canManage
              }
              isSecondaryButtonDisabled={!formProps.isDirty}
              onSecondaryButtonClick={formProps.handleReset}
              onPrimaryButtonClick={formProps.submitForm}
              labelPrimaryButton={FormModalPage.Intl.save}
              labelSecondaryButton={FormModalPage.Intl.revert}
            >
              {loading && (
                <Spacings.Stack alignItems="center">
                  <LoadingSpinner />
                </Spacings.Stack>
              )}
              {error && (
                <ContentNotification type="error">
                  <Text.Body>
                    {intl.formatMessage(messages.channelDetailsErrorMessage)}
                  </Text.Body>
                </ContentNotification>
              )}
              {channel && formProps.formElements}
              {channel && <ApplicationPageTitle additionalParts={[channelName]} />}
              {channel === null && <PageNotFound />}
            </FormModalPage>
          );
```

**Review by emmenko — COMMENTED** [review]
Thanks!

See inline comments for simplify a bit the setup.

Also, it would be nice to also test this functionality via Cypress, since we already have E2E tests for the starter templates 🤗

File: cypress/e2e/template-starter/channels.cy.ts [hunk 3]
```diff
@@ -10,7 +10,8 @@ describe('Channels', () => {
       cy.findByText('Channels list').should('exist');
     });
     cy.findByText('Store Berlin').should('exist');
-    cy.findByText('Store Munich').should('exist');
+    cy.findByText('Store Munich').should('exist').click();
+    cy.findByText('Store Munich - Template starter - mc-e2e-app-kit-01 - Merchant Center').should('exist');
```
Comment (emmenko) [inline cypress/e2e/template-starter/channels.cy.ts:14]: Did you check using `cy.title`?  https://docs.cypress.io/api/commands/title#Examples

```suggestion
    cy.title().should('eq', 'Store Munich - Template starter - mc-e2e-app-kit-01 - Merchant Center');
```

Comment (ddouglasz) [reply cypress/e2e/template-starter/channels.cy.ts]: Updated here: https://github.com/commercetools/merchant-center-application-kit/pull/2804/commits/5601a8028f7a07257709dbf46fc10716bcbaf628
Thank you.

**Review by emmenko — APPROVED** [review]

**Review by kark — APPROVED** [review]
Apart from linter errors, looks good to me 👍

**Review by tdeekens — APPROVED** [review]

**Review by Rhotimee — APPROVED** [review]

**Pull request merged**